src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -729,6 +726,58 @@ private void remove(Resident resident) {
 		TownyUniverse.getInstance().getDataSource().saveTown(this);
 	}
 
+	/**
+	 * Tries to find a new mayor from among the town's residents with the rank specified.
+	 * 
+	 * @param rank - the rank being checked for potential mayors
+	 * @return found - whether or not a new mayor was found
+	 */
+	private boolean findNewMayor(String rank) {
+		boolean found = false;
+		System.out.println("getRank(rank = " + rank + ") = " + getRank(rank));
+		for (Resident newMayor : new ArrayList<>(getRank(rank))) {
```

Comment:
There's no need for an array copy here `new ArrayList<>(getRank(rank))` should just be `getRank(Rank)`

Reply:
I believe this is how the corresponding loop was coded before I rewrote the mayoral succession system.

Reply:
This should be resolved as of 9606f94d6c95ec81a934e87eca2e5ac72278463d.